taichi/ir/analysis.h

```diff
@@ -106,7 +106,8 @@ gather_snode_read_writes(IRNode *root);
 std::vector<Stmt *> gather_statements(IRNode *root,
                                       const std::function<bool(Stmt *)> &test);
 void gather_uniquely_accessed_bit_structs(IRNode *root, AnalysisManager *amgr);
-std::unordered_map<const SNode *, GlobalPtrStmt *>
+std::pair<std::unordered_map<const SNode *, GlobalPtrStmt *>,
```

Comment:
We may need to clarify here: the first component means that the **SNode** is uniquely accessed, but the second component means that the external pointer is uniquely accessed. If I'm not missing something, two `ExternalPtrStmt`s pointing to the same external array doesn't affect whether they are gathered in this function or not. However, two `GlobalPtrStmt`s pointing to the same SNode makes it not uniquely accessed in struct-fors because we are parallel accessing them in the struct-for loop.